src/org/opensolaris/opengrok/history/HistoryGuru.java

```diff
@@ -629,9 +629,7 @@ public void createCache(Collection<String> repositories) {
     public List<Repository> clearCache(Collection<String> repositories) throws HistoryException {
         List<Repository> repos = getReposFromString(repositories);
         HistoryCache cache = historyCache;
-        if (cache == null) {
```

Comment:
I think this can't happen - the `historyCache` is created in `HistoryGuru` constructor.